US22 - Tags administration, fix #30

davidbgk opened the pull request:
ETQ éditrice je veux gérer une page tag afin que les visiteurs puissent consulter les articles par thèmes

- [x] - doit permettre d'ajouter une image et/ou une description (markdown) pour personnaliser une page tag
- [x] - doit permettre de modifier un tag

davidbgk committed:
US22 - Tags administration, fix #30

davidbgk committed:
Merge branch '126-tags-browsability' into 30-tags-administration

davidbgk committed:
Merge branch '126-tags-browsability' into 30-tags-administration

vinyll commented on cafebabel/articles/tags/models.py:
```diff
@@ -39,6 +39,18 @@ def get_images_url(self):
     def get_images_path(self):
         return current_app.config.get('TAGS_IMAGES_PATH')
 
+    def save_from_request(self, request):
```
I know we have this for article too and I don't feel so confident about this in the model; it looks more like a view thing.
Let's keep this for now and brain process it for later ;)

davidbgk replied:
Agreed, it's very fragile and we have to change it during the next sprint. Maybe using WTForms?

vinyll replied:
[Issue open](https://github.com/cafebabel/cafebabel.com/issues/157)

vinyll commented on cafebabel/articles/tags/views.py:
```diff
@@ -36,3 +37,16 @@ def detail(slug):
+
+    if request.method == 'POST':
+        tag.save_from_request(request)
+        flash('Your tag was successfully saved.')
```
This reminds me we are not dealing with errors

vinyll commented on cafebabel/templates/articles/tags/edit.html:
```diff
@@ -0,0 +1,41 @@
+{% extends "base.html" %}
+
```
We may want to add some basic metas (title only?)

davidbgk replied:
We don't need metas on edit pages.

vinyll replied:
I know we don't need SEO but the title might be an indication for history and title bar.
Not big priority though I guess.

davidbgk replied:
Same apply to all edit pages, transversal issue?

vinyll replied:
I suppose it is transverse, but as we are working in // on the meta data it may be why it pops out now.
Not a blocker though.

davidbgk replied:
There are already way too much things to fix within that sprint.

vinyll commented on cafebabel/articles/tags/views.py:
```diff
@@ -36,3 +37,16 @@ def detail(slug):
+@tags.route('/<slug>/edit/', methods=['get', 'post'])
+@editor_required
+def edit(slug):
+    tag = Tag.objects.get_or_404(slug=slug)
```
Won't it break if we have the same slug in 2 languages? ie: _meetmyhood_ that will remain the same throughout languages

johangiraud replied:
domain name (.fr, .en, .de..) will do so it's not the same slug, won't it ?

vinyll replied:
Indeed, to select the slug based on the domain we need to filter it out. That's what I [suggested in this commit](https://github.com/cafebabel/cafebabel.com/pull/153/files#diff-a0088a31e32d86af81efe546d99ad4abR45).

johangiraud replied:
I am a bit lost about it now, let's talk about it later @vinyll

vinyll committed:
tag editing now takes the current language

vinyll commented on cafebabel/articles/tags/views.py:
```diff
@@ -36,3 +37,16 @@ def detail(slug):
+@tags.route('/<slug>/edit/', methods=['get', 'post'])
+@editor_required
+def edit(slug):
+    tag = Tag.objects.get_or_404(slug=slug, language=current_language())
```
I'd like to attract your attention on this @davidbgk.
This solves the issue when there are [twice the same slug in a different language](https://github.com/cafebabel/cafebabel.com/pull/153/files#diff-0896e0b2368aabe52dcdedab894926a3R106).

However this means that changing the language of the current slug would not make it available when redirecting after save.
Nevertheless, I understood that the expected behavior was to browse/work in the language version that you would be reading/editing.

It may be interesting to exchange about this.

davidbgk replied:
If I understand correctly [latest comment](https://github.com/cafebabel/cafebabel.com/issues/30#issuecomment-358245145) from @cafebabel-webmaster we should not be able to change the name, slug of language for a given tag anyway.

davidbgk replied:
OK, now I understand the problem. Thanks for the fix.

johangiraud committed:
styles detail tag page + parallax on picture

johangiraud committed:
clean up test css values

vinyll commented on cafebabel/static/css/styles.css:
```diff
@@ -1454,6 +1454,53 @@ textarea#body {
+}
+
+.tag-page .tag-header-image + header {
+  /*margin-top: 60vh;*/
```
delete this

vinyll commented on cafebabel/static/js/parallax.js:
```diff
@@ -0,0 +1,12 @@
+window.addEventListener('scroll', () => {
+  const scrolledHeight = window.pageYOffset
+  const limit = parallax.offsetTop + parallax.offsetHeight
+  console.log('scrolledHeight: ', scrolledHeight, 'limit: ', limit)
```
remove this

vinyll commented on cafebabel/static/js/parallax.js:
```diff
@@ -0,0 +1,12 @@
+  console.log('scrolledHeight: ', scrolledHeight, 'limit: ', limit)
+  if (scrolledHeight > parallax.offsetTop && scrolledHeight <= limit) {
+    parallax.style.backgroundPositionY =
+      (parallax.offsetTop - scrolledHeight) / 3 + 'px'
```
Template literal?
```js
parallax.style.backgroundPositionY = `${(parallax.offsetTop - scrolledHeight) / 3}px`
```

johangiraud replied:
oh yeah !

vinyll commented on cafebabel/templates/articles/tags/detail.html:
```diff
@@ -1,25 +1,29 @@
 {% extends "base.html" %}
 
+{% block extra_js %}<script src={{ url_for('static', filename='js/parallax.js') }}></script>{% endblock %}
```
this parallax is very specific to tags. It may be interesting to rename it accordingly (_tag-parallax.js_?)

johangiraud replied:
hum ok !

johangiraud committed:
wip styles edit tag page

johangiraud committed:
cleaning after review + change to more precise js file name

johangiraud committed:
styles article list

davidbgk commented on cafebabel/templates/articles/tags/detail.html:
```diff
@@ -1,20 +1,29 @@
 {% block body %}
   {% if tag.has_image %}
-    <img src="{{ tag.image_url }}" alt="Image for {{ tag.name }}">
+    <div class=tag-header-image style="background:url({{ tag.image_url }}) no-repeat center 0 fixed"></div>
```
I'm strongly against that change :)

My take is: do whatever you want with JS but keep at least native HTML elements please.

johangiraud replied:
ok, @pointbar sgonna show me how to do else with js, no prob 👌

johangiraud committed:
add specific metas for tag page

johangiraud committed:
REMOVE add specific metas for tag page

johangiraud committed:
REMOVE styles article list

johangiraud committed:
REMOVE cleaning after review + change to more precise js file name

johangiraud committed:
REMOVE wip styles edit tag page

johangiraud committed:
REMOVE styles detail tag page + parallax on picture

johangiraud committed:
removing meta block and edit button

johangiraud committed:
put back edition button

davidbgk committed:
Merge branch '126-tags-browsability' into 30-tags-administration

# Conflicts:
#	cafebabel/templates/articles/tags/detail.html

davidbgk committed:
Prevent the edition of tag name and language

Indirectly the slug becomes permanent too which solves a couple of nasty side-cases (redirections and so on)

davidbgk committed:
Merge branch '126-tags-browsability' into 30-tags-administration

davidbgk committed:
Merge branch '126-tags-browsability' into 30-tags-administration

# Conflicts:
#	cafebabel/articles/tags/views.py

johangiraud committed:
Merge branch '126-tags-browsability' into 30-tags-administration

johangiraud committed:
move edit button below tag title

johangiraud committed:
Merge branch '30-tags-administration' of github.com:cafebabel/cafebabel.com into 30-tags-administration
